lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -74,13 +74,18 @@ DefConstructor('\documentstyle OptionalSemiverbatim SkipSpaces Semiverbatim []',
     my $class   = ToString($whatsit->getArg(2));
     my $options = $whatsit->getArg(1);
     $options = [($options ? split(/\s*,\s*/, ToString($options)) : ())];
-    # Watch out; In principle, compatibility mode wants a .sty, not a .cls!!!
-    # But, we'd prefer .cls, since we'll have better bindings.
-    # And in fact, nobody's likely to write a binding for a .sty that wants to be a class anyway.
-    # So, we'll just try for a .cls, punting to OmniBus if needed.
-    # If we start wanting to read style files by default, we'll still need to handle this
-    # specially, since class (or sty files pretending to be) cover so much more.
-    if (FindFile($class, type => 'cls', notex => !LookupValue('INCLUDE_CLASSES'))) {
+# Watch out; In principle, compatibility mode wants a .sty, not a .cls!!!
+# But, we'd prefer .cls, since we'll have better bindings.
+# And in fact, nobody's likely to write a binding for a .sty that wants to be a class anyway.
+# So, we'll just try for a .cls, punting to OmniBus if needed.
+# If we start wanting to read style files by default, we'll still need to handle this
+# specially, since class (or sty files pretending to be) cover so much more.
+#
+# EXCEPTION: of course there is an exception. Older arXiv articles use \documentstyle{aipproc}
+# to load a different aipproc.sty than the later evolved aipproc.cls. And of course we need to support that.
+    if ($class eq 'aipproc') {
+      RequirePackage('aipproc', options => $options); }
```

Comment:
There is certainly better general treatment to be invented here, but which?? I didn't want to get in the way of loading `revtex.cls` when encountering `\documentstyle{revtex}` by reversing the default... And since so far this is our only exception, I excepted it.

Reply:
The general comment about loading .cls was probably more true originally. I suspect that the few cases where we have both a sty and cls (aastex,elsart,emulateapj,revtex,revtex4,subfiles) have gradually been (re)written into a consistent style of both cases loading a support file, and the .cls files also loading article.  So, perhaps the following structure would be better and avoid special casing:
```
if(FindFile($class, type=>'pkg'...){
   InputDefinitions('article', type=>'cls'...
   RequirePackage($class...
elsif(FindFile($class, type=>'cls'...
   LoadClass($class,...
else {
  InputDefinitions('OmniBus'...
```

Reply:
Cool, we can try being more ambitious!
I wonder about the mandatory preload of article.cls though, shouldn't that be the responsibility of whichever "legacy" .sty file we are loading? e.g. in this aipproc.sty case we'd load revtex.cls instead of article, from within that legacy package binding.

In a way, we're either ready to do it in general, or we aren't. I thought we weren't so I made the exception, but if any of the .sty files you enumerated are to follow the same treatment, they should be able to imply a LoadClass call, right? (aastex,elsart,emulateapj,revtex,revtex4,subfiles).

Reply:
Taking the case of `aastex`, based on passed options the `aastex.cls` is able to either load article.cls or revtex4.cls. Meanwhile the `aastex.sty` simple loads the `_support` binding without taking care of the class options. 

So changing the dispatch would break the revtex4 portion of aastex support that the current code is able to load, not ideal. Hm...

Reply:
The way I see it the current rule is mostly correct, but it has exceptions. I can phrase it as: "whenever latexml has bindings for X.sty and X.cls, as a rule the X.cls is a related superset over X.sty, and can be used directly for \documentstyle".

The issue with aipproc.sty and aipproc.cls is that they are "too distant" and have breaking changes, so they're better thought of as separate entities that aren't as interchangeable as the other cases we have. Maybe we need a way of marking those cases up, but keep the default expectation that "X.cls is compatible with X.sty"?